nistats/second_level_model.py

```diff
@@ -0,0 +1,381 @@
+from warnings import warn
+import sys
+import time
+import pandas as pd
+import numpy as np
+from nibabel import Nifti1Image
+
+from sklearn.base import BaseEstimator, TransformerMixin, clone
+from nilearn._utils.niimg_conversions import check_niimg
+from nilearn._utils import CacheMixin
+from nilearn.input_data import NiftiMasker
+from patsy import DesignInfo
+
+from .first_level_model import FirstLevelModel
+from .first_level_model import run_glm
+from .regression import OLSModel, SimpleRegressionResults
+from .contrasts import compute_contrast
+from .utils import _basestring
+from .design_matrix import create_second_level_design
+
+
+class SecondLevelModel(BaseEstimator, TransformerMixin, CacheMixin):
+    """ Implementation of the General Linear Model for multiple subject
+    fMRI data
+
+    Parameters
+    ----------
+
+    mask: Niimg-like, NiftiMasker or MultiNiftiMasker object, optional,
+        Mask to be used on data. If an instance of masker is passed,
+        then its mask will be used. If no mask is given,
+        it will be computed automatically by a MultiNiftiMasker with default
+        parameters. Mask automatic computation assumes first level imgs have
+        already been masked.
+
+    smoothing_fwhm: float, optional
+        If smoothing_fwhm is not None, it gives the size in millimeters of the
+        spatial smoothing to apply to the signal.
+
+    memory: string, optional
+        Path to the directory used to cache the masking process and the glm
+        fit. By default, no caching is done. Creates instance of joblib.Memory.
+
+    memory_level: integer, optional
+        Rough estimator of the amount of memory used by caching. Higher value
```

Comment:
This docstring is a bit hard to follow. When I read "Rough estimator of the amount of memory used by caching" I start thinking "My data are bout 300MB", which is not what is expected. The expected value is typically 1.